### PR TITLE
Update main clvm document.

### PR DESCRIPTION
**Pull request by aqk**
* removed "0 also evaluates to the root of the argument tree"
* changed behaviour: opcodes never accept more arguments than required (ignore extra args)
* add nil is self-quoting
* added `/`
* updated to use new quote syntax
* Removed opcodes tables to another PR.

### CODE REVIEW

File: docs/ref/clvm.md [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+* **ChiaLisp** - A higher-level language, built on top of CLVM.
+* **CLVM Object** - The underlying data type in the CLVM. An atom or a cons pair.
+* **Atom** - The datatype for values in the CLVM. Atoms are immutable byte arrays. Atoms are untyped and are used to encode all strings, integers, and keys. The only things in the CLVM which are not atoms are cons pairs. Atom properties are length, and the bytes in the atom.
+* **cons pair** - An immutable ordered pair of references to other CLVM objects. One of two data types in the CLVM. The syntax for a cons pair is a dotted pair. Also called `cons cell` or `cons box`.
+* **slot** - One of the cells in a cons box. right or left. Accessed with `f` (first) or `r` (rest).
+* **nil** - nil is the special value represented by the zero length byte array. This value represents zero, the empty string, false, and the empty list. Nil is represented in CLVM assembly as `()`, `0`, or `''`.
+* **Value** - We use value to mean an abstract value like `1` (an integer), `0xCAFE` (a byte string), `"hello"` (a string) or `(sha256 (q . "hello"))` (a program). Values are represented by CLVM Objects.
```
Comment (matt-o-how) [inline docs/ref/clvm.md:20]: It's probably worth mentioning that Values are Atoms

File: docs/ref/clvm.md [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+(substr (q . "clvm") (q . 4) (q . 5)) => FAIL
+(substr (q . "clvm") (q . 1) (q . 0)) => FAIL
+(substr (q . "clvm") (q . -1) (q . 4)) => FAIL
+```
+
+**strlen** `(strlen S)` return the number of bytes in S.
+
+```
+(strlen (q . "clvm")) => 4
+(strlen (q . "0x0")) => 1
```
Comment (matt-o-how) [inline docs/ref/clvm.md:362]: This is not correct.

(strlen (q . 0x0)) => 1
(strlen (q . "0x0")) => 3

File: sidebars.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   someSidebar: {
     Chialisp: ['doc1', 'doc2', 'doc3', 'doc4', 'doc5'],
-    CLVM: ['ref/serialization', 'ref/opcodes'],
+    CLVM: ['ref/clvm'],
```
Comment (aqk) [inline sidebars.js:4]: I'll re-add serialization here - there are 2 more PRs for the opcode tables.

**Review by hoffmang9 — APPROVED** [review]
+1

**Pull request merged**